gzip.go

```diff
@@ -63,35 +69,93 @@ func addLevelPool(level int) {
 // GzipResponseWriter provides an http.ResponseWriter interface, which gzips
 // bytes before writing them to the underlying response. This doesn't close the
 // writers, so don't forget to do that.
+// It can be configured to skip response smaller than minSize.
 type GzipResponseWriter struct {
 	http.ResponseWriter
 	index int // Index for gzipWriterPools.
 	gw    *gzip.Writer
+
+	code int // Saves the WriteHeader value.
+
+	minSize      int    // Specifed the minimum response size to gzip. Only the first Write call is checked.
+	buff         []byte // Holds the first part of the write before the reaching the minSize or the end of the write.
+	bytesWritten int    // Keep trace of the numbers of bytes written.
 }
 
 // Write appends data to the gzip writer.
 func (w *GzipResponseWriter) Write(b []byte) (int, error) {
```

Comment:
Your are is right.
This function is now splitted into two cleaner and shorter functions.

Does it look better to you?